crates/ruff_server/src/server.rs

```diff
@@ -73,10 +73,10 @@ impl Server {
 
         Ok(Self {
             conn,
-            client_capabilities,
             threads,
             worker_threads,
-            session: Session::new(&server_capabilities, &workspaces)?,
+            session: Session::new(&client_capabilities, &server_capabilities, &workspaces)?,
+            client_capabilities,
```

Comment:
Would it be sufficient to only store the client capabilities on the session, or why do we need to store it on the sessoin and server?

Reply:
We need a separate reference to `client_capabilities` to pass into `Server::try_register_capabilities` because the session is already borrowed mutably by the scheduler.